setup.py

```diff
@@ -29,6 +29,7 @@
     install_requires=[
         'click',
         'setuptools',
+        'pip'
```

Comment:
What's the scenario being solved for here? Is there an expectation that one might perform the initial install via Conda or another package manager?

Reply:
I updated the description to give more context.  EDS could be installed by something besides pip, but internally, EDS will use pip to check what's installed.

Reply:
Not to be overly pedantic... but AFAIK pip has been included with Python since 3.4. If we require py >= 3.7, shouldn't this always be met?

Reply:
debian/ubuntu doesn't bundle `pip` or the [ensurepip](https://docs.python.org/3/library/ensurepip.html) module

Reply:
> debian/ubuntu doesn't bundle `pip` or the [ensurepip](https://docs.python.org/3/library/ensurepip.html) module

I try not to swear on public github, so I'm not quite sure what to say. Ok, good to know, thanks. I sit corrected.

Reply:
one alternative I considered is that EDS would not do any installation checks.   the main routine would just delegate the installation builds out to the worker plugin regardless, then the worker deals with it.   For example, an AWS worker implementation, would likely be running EDS builds using code build, and would exit quickly if requirements were already met.   The worker would relaunch EDS executions and set attributes on the Event object as to whether EDS was built or not.  You can see in https://github.com/qwcode/eds/pull/1 this was what I was originally thinking, but then decided to convert over to using these pip internal checks.   If using the pip internals turns out to be buggy, we can consider the other plan.